defaults/main.yml

```diff
@@ -0,0 +1,47 @@
+# SPDX-License-Identifier: MIT
+---
+# The provider for nbde_client role. Currently 'clevis' is supported.
+nbde_client_provider: clevis
+
+# nbde_client_bindings is a list, and its elements support the following keys:
+#
+# * device             - specifies the path of a LUKS device. (REQUIRED)
+# * pass               - specifies a valid passphrase for this device.
+# * keyfile            - specifies either the absolute or relative path of a
+#                        valid keyfile for the device pointed out by device.
```

Comment:
On the managed host or on the control node?

Reply:
> > Does this look good to everyone? All we need is better text for the `passphrase_temporary`?
> 
> First of all, the code and examples need changing, they are still using `discard_passphrase`

discard_passphrase was not changed yet, so this is expected, but the examples in the library obsolete, as you pointed out. Let me update them.

Reply:
> On the managed host or on the control node?

on the control node. Updating this as well.

Reply:
> > On the managed host or on the control node?
> 
> on the control node. Updating this as well.

That's too bad, because the same parameter in the storage role specifies a path on the managed node... https://github.com/linux-system-roles/storage/pull/102#discussion_r439003264 is there a good reason for this difference?

Reply:
> > > On the managed host or on the control node?
> > 
> > 
> > on the control node. Updating this as well.
> 
> That's too bad, because the same parameter in the storage role specifies a path on the managed node... [linux-system-roles/storage#102 (comment)](https://github.com/linux-system-roles/storage/pull/102#discussion_r439003264) is there a good reason for this difference?

Actually, for us the `device` is on the managed host and the `keyfile` is on the control node, and we would copy it ourselves. From your comment, I thought you were referring to the `keyfile`.

I don't feel strongly about requiring the keyfile already to be on the managed host. In fact, it would make things simpler for us. @richm, what do you think here?

Reply:
(disclaimer: quite possibly on the control node makes more sense, I just dislike the inconsistency in semantics.)

Reply:
> I don't feel strongly about requiring the keyfile already to be on the managed host. In fact, it would make things simpler for us. @richm, what do you think here?

How will the file get to the `keyfile` location on the managed host?

Reply:
If we require it to be on the managed host, it would be the admin's responsibility to get it there. I think what we are doing now (copying it ourselves) makes sense.

Reply:
@sergio-correia 
> From your comment, I thought you were referring to the keyfile

Yes, I was referring to the keyfile.

> I don't feel strongly about requiring the keyfile already to be on the managed host. In fact, it would make things simpler for us. 

Important is what is simpler for the user, and I have no idea how keyfiles are used in practice. Are they usually lying on the managed host? By the way, in storage we seem to be leaning towards having keyfile on the control node as well. Can you please help them with the implementation?